planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -437,6 +436,8 @@ void GoalPlannerModule::onFreespaceParkingTimer()
     needPathUpdate(
       local_planner_data->self_odometry->pose.pose, path_update_duration, modified_goal_opt,
       parameters)) {
+    auto goal_searcher = std::make_shared<GoalSearcher>(parameters, vehicle_footprint);
+
```

Comment:
```suggestion
    const auto goal_searcher = std::make_shared<GoalSearcher>(parameters, vehicle_footprint);

```

Reply:
Since it is now only used within planFreespacePath(), it may be ok to generate it within that function.

Reply:
(update() does not depend on search(), and there is no problem if we pass goal candidates. It seems like it would be ok to make update() a static method.)